### ISSUE TITLE
npm install srp error

### ISSUE THREAD

**Issue by ghost** [homebridge/HAP-NodeJS#305]
`npm WARN package.json node@0.0.0 No description
npm WARN package.json node@0.0.0 No repository field.
npm WARN package.json node@0.0.0 No README data
npm WARN engine deep-extend@0.4.1: wanted: {"node":">=0.12.0","iojs":">=1.0.0"} (current: {"node":"0.10.29","npm":"1.4.21"})
npm WARN engine hawk@3.1.3: wanted: {"node":">=0.10.32"} (current: {"node":"0.10.29","npm":"1.4.21"})
npm WARN engine form-data@2.0.0: wanted: {"node":">= 0.12"} (current: {"node":"0.10.29","npm":"1.4.21"})
npm WARN engine cryptiles@2.0.5: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.29","npm":"1.4.21"})
npm WARN engine hoek@2.16.3: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.29","npm":"1.4.21"})
npm WARN engine boom@2.10.1: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.29","npm":"1.4.21"})
\

> bignum@0.12.5 install /home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum
> node-pre-gyp install --fallback-to-build

node-pre-gyp ERR! Tried to download: https://rvagg-node.s3-us-west-2.amazonaws.com/bignum/v0.12.5/bignum-v0.12.5-node-v11-linux-arm.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for bignum@0.12.5 and node@0.10.29 (node-v11 ABI) (falling back to source compile with node-gyp)
gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/0.10.29"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/.node-gyp"
make: Entering directory '/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/build'
make: **\* No rule to make target '../.node-gyp/0.10.29/common.gypi', needed by 'Makefile'.  Stop.
make: Leaving directory '/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/pi/HAP-NodeJS/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:809:12)
gyp ERR! System Linux 4.4.21-v7+
gyp ERR! command "/usr/bin/nodejs" "/home/pi/HAP-NodeJS/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/binding/bignum.node" "--module_name=bignum" "--module_path=/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/binding"
gyp ERR! cwd /home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum
gyp ERR! node -v v0.10.29
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/nodejs /home/pi/HAP-NodeJS/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/binding/bignum.node --module_name=bignum --module_path=/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/binding' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:755:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:822:5)
node-pre-gyp ERR! System Linux 4.4.21-v7+
node-pre-gyp ERR! command "node" "/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum
node-pre-gyp ERR! node -v v0.10.29
node-pre-gyp ERR! node-pre-gyp -v v0.6.30
node-pre-gyp ERR! not ok
Failed to execute '/usr/bin/nodejs /home/pi/HAP-NodeJS/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/binding/bignum.node --module_name=bignum --module_path=/home/pi/HAP-NodeJS/node_modules/srp/node_modules/bignum/binding' (1)
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian

npm ERR! bignum@0.12.5 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bignum@0.12.5 install script.
npm ERR! This is most likely a problem with the bignum package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls bignum
npm ERR! There is likely additional logging output above.

npm ERR! System Linux 4.4.21-v7+
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "install" "srp"
npm ERR! cwd /home/pi/HAP-NodeJS
npm ERR! node -v v0.10.29
npm ERR! npm -v 1.4.21
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /home/pi/HAP-NodeJS/npm-debug.log
npm ERR! not ok code 0
`

**Comment by oriolism** [comment]
Im also having issues installing srp. Did you manage to fix your problem?

**Comment by TheNoim** [comment]
I also had this issue, but i can't remember how i fixed this. Maybe try npm install npm@next

**Comment by warloofer** [comment]
I had this problem and I solved it uninstalling Nodejs completely and installing it with the installer of nodejs: _https://nodejs.org/en/download/package-manager/_

Another thing I did, was to install only the packages that Node requested me when trying to launch it (node Core.js). I found that node doesnt require SRP, instead of it, request me the package "fast-srp-hap". 

Of course, you will need to install "apt-get install libavahi-compat-libdnssd-dev" before all node installation, and remember to update your rasbpian (apt-get update, sudo apt-get upgrade -y).
